.circleci/verbatim-sources/job-specs-setup.yml

```diff
@@ -0,0 +1,12 @@
+  setup:
+    docker:
+      - image: circleci/python:3.7.3
+    steps:
+      - checkout
```

Comment:
This replaces the location of the scripts for all the binaries from this checkout instead of https://github.com/pytorch/pytorch/blob/master/.circleci/scripts/binary_checkout.sh#L21-L41 . Will they always be the same?

Reply:
For context, I added our own checkout for all our binary jobs cause they check out two repos, not just one, but I wasn't trying to do anything magical.

Reply:
The checkout change wasn't intended to be anything special - the main thing we did was use the workspace to share the scripts around.

The setup job persists the scripts that any job will need into the workspace, and then all the other jobs can pick them up from there instead of having to `checkout` or `curl` with the retry.

Reply:
I don't think the files will always be the same. For example this PR https://github.com/pytorch/pytorch/pull/19765 would start a bunch of jobs that checkout an arbitrary pytorch branch. By changing binary_checkout.sh we can point to that branch directly. But if all the jobs use the built-in checkout then they would all use the same commit the CircleCI is running on.

Reply:
I'm not quite following what the goal is there, if you wanted to run against an arbitrary branch, can you open a pull request for that branch to have it start building?

CircleCI usually builds every branch unless there's a branch filter in the workflow config, but this project is configured to only build when a PR exists, see "Advanced Settings":
>By default, we will build all the commits for this project. Once turned on, we will only build branches that have associated pull requests open. 

<hr />

We haven't intentionally changed to not use binary_checkout in any of the "real" jobs, we only added `checkout` to the new `setup` job, which is used to provide the `.circleci/scripts` directory to the other jobs.

The scripts directory that gets passed around in `~/workspace` would be the one from the PR branch, but the rest of the jobs remain free to do whatever they would like.

Reply:
We haven't currently changed any of the existing steps which use curl to download a script file - we've only used the workspace mechanism for files which were previously inlined. Based on my understanding of how things fit together, I believe this approach should be a suitable replacement for every scenario where you currently curl a file down from github if you wanted to move over.

Reply:
I am following this up in another comment below.